src/elfcore.c

```diff
@@ -1196,8 +1196,8 @@ static int CreateElfCore(void *handle,
         /* Align all following segments to multiples of page size            */
         if (note_align) {
           char scratch[note_align];
-          memset(scratch, 0, sizeof(scratch));
-          if (writer(handle, scratch, sizeof(scratch)) != sizeof(scratch)) {
+          memset(scratch, 0, note_align*sizeof(char));
```

Comment:
According to c++ std, sizeof(char) is alway 1 on any platform, so it can be omitted.